docs/005-mobile-app-architecture.adoc

```diff
@@ -0,0 +1,153 @@
+= ADR 005 - Mobile App Architecture
+Daniel Karzel <daniel@10101.finance>
+3.0, July 29, 2022: AsciiDoc article template
+:toc:
+:icons: font
+:attributes: 2023-02-02
+
+This document contains the reasoning for high level architecture and project structure decisions for the mobile app.
+
+== Decision
+
+* Application State and data storage are handled in the mobile app's native rust code.
+** Whenever data is required the app should ask the native backend for data.
+** Ephemeral (widget) state should be kept on the presentation layer.
+* The Dart presentation layer is encapsulated as presentation layer
+** We have a Dart and a rust model and map between them
+** No rust ffi code should be use in the presetnation layer, i.e. no ffi includes in `ChangeNotifier`s and `Widget`s.
+** Ffi bindings should be encapsulated in `Service`s in Dart
```

Comment:
nit: consider using `FFI`  (here and other spots) - it's an acronym, just like API and other acronyms that are uppercase in this doco